### PR TITLE
Add new programs to learning and exercises packages

### PR DESCRIPTION
**Pull request by marekk1992**

### CODE REVIEW

File: src/exercises/sharedDigit/SharedDigit.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package exercises.sharedDigit;
+
+public class SharedDigit {
+
+    public static boolean hasSharedDigit(int firstNumber, int secondNumber) {
+        if (firstNumber < 10 || firstNumber > 99 || secondNumber < 10 || secondNumber > 99) {
```
Comment (RobertasVainickas) [inline src/exercises/sharedDigit/SharedDigit.java:6]: Would be nice if this logic could be placed in separate method like you did before.

File: src/exercises/sharedDigit/SharedDigit.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            return false;
+        }
+
+        int digit;
```
Comment (RobertasVainickas) [inline src/exercises/sharedDigit/SharedDigit.java:10]: Should be declared inside of the loop

**Review by RobertasVainickas — CHANGES_REQUESTED** [review]

File: src/exercises/evenDigitSum/EvenDigitSum.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+public class EvenDigitSum {
+
+    public static int getEvenDigitSum(int number) {
+        if (number < 0) {
+            return -1;
+        }
+
+        int sum = 0;
+        int digit;
```
[the remark this answers is not in the document]
Comment (marekk1992) [reply src/exercises/evenDigitSum/EvenDigitSum.java]: Fixed.

File: src/exercises/firstLastDigitSum/FirstLastDigitSum.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+public class FirstLastDigitSum {
+
+    public static int sumFirstAndLastDigit(int number) {
+        if (number < 0) {
+            return -1;
+        }
+
+        if (number < 10) {
+            return number * 2;
+        } else {
```
[the remark this answers is not in the document]
Comment (marekk1992) [reply src/exercises/firstLastDigitSum/FirstLastDigitSum.java]: Fixed.

File: src/exercises/sharedDigit/SharedDigit.java (hunk 1 above)
Comment (marekk1992) [reply src/exercises/sharedDigit/SharedDigit.java]: Introduced new method.

File: src/exercises/sharedDigit/SharedDigit.java (hunk 2 above)
Comment (marekk1992) [reply src/exercises/sharedDigit/SharedDigit.java]: Fixed.

**Review by RobertasVainickas — APPROVED** [review]

**Pull request merged**